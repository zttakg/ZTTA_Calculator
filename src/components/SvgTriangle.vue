<template>
  <SvgPolygon :points="points" :fillColor="fillColor" />
</template>

<script>
import { SvgPolygon } from "@/components";

export default {
  name: "SvgTriangle",

  components: {
    SvgPolygon
  },

  props: {
    side1: {
      type: String,
      default: "300"
    },
    side2: {
      type: String,
      default: "300"
    },
    fillColor: String
  },

  computed: {
    points() {
      return `0,0 0,${this.side1} ${this.side2},${this.side1} 0,0`;
    }
  }
};
</script>
