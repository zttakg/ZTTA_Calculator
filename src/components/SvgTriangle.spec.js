import { mount } from "@vue/test-utils";
import SvgTriangle from "./SvgTriangle";

describe("<SvgTriangle /> component", () => {
  it("should render correct without props", () => {
    const wrapper = mount(SvgTriangle);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = mount(SvgTriangle, {
      propsData: {
        side1: "200",
        side2: "400",
        fillColor: "#000"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
