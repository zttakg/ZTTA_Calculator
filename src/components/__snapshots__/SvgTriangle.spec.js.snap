// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgTriangle /> component should render all props 1`] = `<polygon points="0,0 0,200 400,200 0,0" fill="#000"></polygon>`;

exports[`<SvgTriangle /> component should render correct without props 1`] = `<polygon points="0,0 0,300 300,300 0,0"></polygon>`;
